### ISSUE TITLE
While running the fastlane beta "block (2 levels) in shift': [!] Illegal quoting in line 1. (CSV::MalformedCSVError)

### ISSUE THREAD

**Issue by nihp** [fastlane/fastlane#12611]
Fastlane beta does not export the build to crashlytics. I got an error as block (2 levels) in shift': [!] Illegal quoting in line 1. (CSV::MalformedCSVError)

Kindly anyone help me to solve this.I does not have any idea to solve this error.

I need to generate a build and upload to Crashlytics to see the crashes inside the app.

```
/.rvm/rubies/ruby-2.5.1/lib/ruby/gems/2.5.0/gems/fastlane-2.95.0/fastlane/lib/fastlane/actions/register_devices.rb:38:in `run'
/.rvm/rubies/ruby-2.5.1/lib/ruby/2.5.0/csv.rb:1346:in `read'
/.rvm/rubies/ruby-2.5.1/lib/ruby/2.5.0/csv.rb:1289:in `open'
    /.rvm/rubies/ruby-2.5.1/lib/ruby/2.5.0/csv.rb:1346:in `block in read'
/.rvm/rubies/ruby-2.5.1/lib/ruby/2.5.0/csv.rb:1777:in `read'
/.rvm/rubies/ruby-2.5.1/lib/ruby/2.5.0/csv.rb:1777:in `to_a'
/.rvm/rubies/ruby-2.5.1/lib/ruby/2.5.0/csv.rb:1763:in `each'
/.rvm/rubies/ruby-2.5.1/lib/ruby/2.5.0/csv.rb:1821:in `shift'
/.rvm/rubies/ruby-2.5.1/lib/ruby/2.5.0/csv.rb:1821:in `loop'
/.rvm/rubies/ruby-2.5.1/lib/ruby/2.5.0/csv.rb:1867:in `block in shift'
/.rvm/rubies/ruby-2.5.1/lib/ruby/2.5.0/csv.rb:1867:in `each'
    /.rvm/rubies/ruby-2.5.1/lib/ruby/2.5.0/csv.rb:1911:in `block (2 levels) in shift': [!] Illegal quoting in line 1. (CSV::MalformedCSVError)
```

My fastfile is like 

```
fastlane_version "2.25.0"

default_platform :ios

platform :ios do
  lane :beta do |values|
    # Before calling match, we make sure all our devices are registered on the Apple Developer Portal
    register_devices(
      devices_file: "./fastlane/devices.txt",
      username: 'user@abc.com'
    )

    # After registering the new devices, we'll make sure to update the provisioning profile if necessary
    # options for type are "adhoc", "development", "appstore"
    match(
      force_for_new_devices: true,
      type: "development",
      # readonly: true  #readonly mode to be sure it won't create any new certificates or profiles.
    )
    
    # Change build number (requirement of Testflight).
    increment_build_number(
     xcodeproj: "./AppName.xcodeproj"
    )

    # Build a release
    gym(
      scheme: "AppName",
      clean: true,
      workspace: "./AppName.xcworkspace" # since using pods
      # project: "./AppName.xcodeproj"
    )

    emails = values[:test_email] ? values[:test_email] : ['user1@abc.com', 'user2@abc.com'] # You can list more emails here
    groups = values[:test_email] ? nil : nil # You can define groups on the web and reference them here

    # Upload release to crashlytics 
    crashlytics(
      api_token: "[insert_key_here]",
      build_secret: "[insert_key_here]",
      emails: emails,
      groups: groups,
      notes: 'Distributed with fastlane', # Check out the changelog_from_git_commits action
      notifications: true # Should this distribution notify your testers via email?
    )

    # You can notify your team in chat that a beta build has been uploaded
    # slack(
    #   slack_url: "https://hooks.slack.com/services/YOUR/TEAM/INFO"
    #   channel: "beta-releases",
    #   message: "Successfully uploaded a beta release - see it at https://fabric.io/_/beta"
    # )
  end
end
```

**Comment by revolter** [comment]
Are you able to post there the contents of the `./fastlane/devices.txt` file?

**Comment by nihp** [comment]
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.1">
<dict>
        <key>Device UDIDs</key>
        <array>
                <dict>
                        <key>deviceIdentifier</key>
                        <string>xxxxxxxxxxxxxxxxxxxxxxxx</string>
                        <key>iPhone X</key>
                        <string>IPHONE NAME</string>
                </dict>
        </array>
</dict>
</plist>


Clarification:

File name for the devices is given as devices in my project not as devices.txt. Is it mandatory to give the devices files as devices.txt?

**Comment by nihp** [comment]
Also I have given the folder as 

Device ID     Device Name
XXXXXXXXXXX     Device name

I don't know how the exact format of my devices

**Comment by revolter** [comment]
_(I never used the Crashlytics action, but I'll try to help you by guessing)_

I think that what you posted it from a `*.deviceids` file, right? From what I see, there is a standard format for this as you can see mentioned in the second example here: https://docs.fastlane.tools/actions/register_devices/#4-examples

**Comment by revolter** [comment]
Is there a `tab` character between `Device ID` and `Device Name` and between `...XXX` and `Device name`?

**Comment by nihp** [comment]
I have given the space

**Comment by joshdholtz** [comment]
@nihp It should be tabs like @revolter had mentioned 😊 (taken from http://devimages.apple.com/downloads/devices/Multiple-Upload-Samples.zip)
```
Device ID	Device Name
A123456789012345678901234567890123456789	NAME1
B123456789012345678901234567890123456789	NAME2
```

**Comment by revolter** [comment]
Replace it/them with tabs instead and let me know if it works.

**Comment by nihp** [comment]
Thanks @revolter 
Thanks for yr reply @joshdholtz 
It works after I have given the devices as you mentioned here. 

https://docs.fastlane.tools/actions/register_devices/#4-examples 

But after running the fastlane beta it showing the error while I am trying to upload the build in the crashlytics.

Here I have run the fastlane beta and gave my repository url and the password which I has given while running the match appstore. But it repeatedly showing the same. 

How can I upload the build automatically in crashlytics? 

Note: I already done the fastlane to upload the build in Testflight. I need to know the crash reports very clearly for individual testers. So I am trying to integrate Crashlytics  with in my project.

```
Enter the passphrase that should be used to encrypt/decrypt your certificates
 This passphrase is specific per repository and will be stored in your local keychain
 Make sure to remember the password, as you'll need it when you run match on a different machine
 Passphrase for Git Repo: 
```

**Comment by joshdholtz** [comment]
@nihp You need to now setup `match` - https://docs.fastlane.tools/actions/match/#match

I"m going to close this issue for now as your CSV issue is solved but you will need to setup each tool/action that you are using in your `beta` lane before things will work 😊 The docs (https://docs.fastlane.tools) and searching through past issues will be your friend 💪

**Comment by revolter** [comment]
I don't know if I understood correctly, but it looks like you want to distribute beta builds view TestFlight and receive crashes via Crashlytics.

For this, you don't need the `crashlytics` action at all. That's for distributing beta builds via their platform, so you should use either Crashlytics either TestFlight for beta distributions.

To only receive crashes you just need to follow the steps in their documentation: https://fabric.io/kits/ios/crashlytics/install

**Comment by nihp** [comment]
I have followed the steps in the https://fabric.io/kits/ios/crashlytics/install. 

My requirement:

**I need to distribute my beta build and also need to know about the crashed logs.** 

I have followed this document to upload the build in testflight https://medium.com/react-native-training/fastlane-for-react-native-ios-android-app-devops-8ca85bee614e.

Here it will upload our build directly to testflight and we can't able to see the crash logs.

Really I need to know the crashed logs from my App. So I have followed this https://medium.com/@tjkangs/react-native-with-fastlane-and-crashlytics-beta-aa0d6ca630fd

I need to know how the build will be uploaded in the crashlytics and how we know about the crashes fastlane and crashlytics. Is this is a right way or not. I don't know.

Kindly let me know your thoughts to know the crashed logs in crashlytics.
